velox/common/encode/tests/Base64Test.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/encode/Base64.h"
+#include <gtest/gtest.h>
+#include "velox/common/base/tests/GTestUtils.h"
+
+namespace facebook::velox::encoding {
+class Base64Test : public ::testing::Test {};
+
+TEST_F(Base64Test, fromBase64) {
+  EXPECT_EQ(
+      "Hello, World!",
+      Base64::decode(folly::StringPiece("SGVsbG8sIFdvcmxkIQ==")));
+  EXPECT_EQ(
+      "Base64 encoding is fun.",
+      Base64::decode(folly::StringPiece("QmFzZTY0IGVuY29kaW5nIGlzIGZ1bi4=")));
+  EXPECT_EQ(
+      "Simple text", Base64::decode(folly::StringPiece("U2ltcGxlIHRleHQ=")));
+  EXPECT_EQ(
+      "1234567890", Base64::decode(folly::StringPiece("MTIzNDU2Nzg5MA==")));
+
+  // Check encoded strings without padding
+  EXPECT_EQ(
+      "Hello, World!",
+      Base64::decode(folly::StringPiece("SGVsbG8sIFdvcmxkIQ")));
+  EXPECT_EQ(
+      "Base64 encoding is fun.",
+      Base64::decode(folly::StringPiece("QmFzZTY0IGVuY29kaW5nIGlzIGZ1bi4")));
+  EXPECT_EQ(
+      "Simple text", Base64::decode(folly::StringPiece("U2ltcGxlIHRleHQ")));
+  EXPECT_EQ("1234567890", Base64::decode(folly::StringPiece("MTIzNDU2Nzg5MA")));
+}
+
+TEST_F(Base64Test, calculateDecodedSizeProperSize) {
+  size_t encoded_size{0};
+
+  encoded_size = 20;
```

Comment:
Perhaps these tests could be rewritten to reduce the copy&paste for more compactness
with something like this
```
std::vector<std::tuple<size_t, std::string, size_t>> testGoodData = {
{ 13, "SGVsbG8sIFdvcmxkIQ==", 18 },
...
}

for (auto [result, encoded, adjustedLength] : testGoodData) {
  size_t length = encoded.length();
  EXPECT_EQ(result, Base64::calculateDecodedSize(encoded, length);
  EXPECT_EQ(adjustedLength, length);
}
```

And then similarly for the error test(s). I see only one, could there be more? If not then it is sufficient as it is.

What happens with empty string?